Added explanation to fmi3{Get|Set}{Interval|Shift}Fraction arguments (#1993)
modelica/fmi-standard: docs/2_2_common_mechanisms.adoc

<<fmi3SetClock>> must not be called in Scheduled Execution, instead <<fmi3ActivateModelPartition>> must be called.

> The Clocks API function argument names <<intervals>>, <<shifts>>, <<counters>> and <<resolutions>> are chosen to match the link:https://specification.modelica.org/maint/3.6/synchronous-language-elements.html#clock-constructors[Modelica Clock Constructors].
> However, the FMI standard does not assume that the importer tool's clock implementation matches the Modelica Clock implementation.
> In general, the arguments <<counters>> and <<resolutions>> have the meaning of a "numerator" and a "denominator".

Reviewer: ```suggestion
_[The Clocks API function argument names <<intervals>>, <<shifts>>, <<counters>> and <<resolutions>> are chosen to match the link:https://specification.modelica.org/maint/3.6/synchronous-language-elements.html#clock-constructors[Modelica Clock Constructors]._
_However, the FMI standard does not assume that the importer tool's clock implementation matches the Modelica Clock implementation._
_In general, the arguments <<counters>> and <<resolutions>> have the meaning of a "numerator" and a "denominator".]_
```
The explanation should be non-normative.

Reviewer 2: ```suggestion
_[The Clocks API function argument names <<intervals>>, <<shifts>>, <<counters>> and <<resolutions>> are chosen to match the naming used in link:https://specification.modelica.org/maint/3.6/synchronous-language-elements.html#clock-constructors[Modelica Clock Constructors]._
_This commonality of naming does not imply any normative reference to the semantics of Modelica._
_Mathematically speaking, the arguments <<counters>> and <<resolutions>> in the fraction variant of these functions serve as numerators and denominators of fractional representations of the corresponding floating-point number arguments of the decimal variant.]_
```

Maybe the final sentence should be split up a bit more, but didn't find the fortune to do so.

* `valueReferences` is an array of size `nValueReferences` holding the value references of the Clock variables.

[[counters,`counters`]]

> * <<counters>> _[Note: This variable may increment by values other than 1.]_ and

Reviewer: Why remove this note?

Author: This variable is never incremented. The implementation in the importer tool may increment some counter, but definitely not the "counter" argument of this function.
Furthermore there also may not be a counter in the importer tool implementation altogether. The FMI does not assume the implementation.

Reviewer 3: Of course it can be increment it, but it can also be change with any increment (or decrement). Isn't that what the note is there to clarify?

Author: ```C
typedef fmi3Status fmi3SetIntervalFractionTYPE(fmi3Instance instance,
                                               const fmi3ValueReference valueReferences[],
                                               size_t nValueReferences,
                                               const fmi3UInt64 counters[],
                                               const fmi3UInt64 resolutions[]);
```
It is const

Reviewer 3: That doesn't prevent the importing tool from changing it between calls to `fmiSetInteralFractionType` (if the `intervalVariability` is `tunable`).

Reviewer 2: Given that I think this note was added after people were confused on the rules,  I don't think removing it is a sensible way to go. We might want to adjust it going forward.

> * <<counters>> and

Reviewer 2: ```suggestion
* <<counters>> _[Note: This variable may increment by values other than 1.]_ and
```
